All instructor pages implemented.

chrisprosan opened the pull request:
Changes:
- Api and database structure are updated
- All Instructor pages are coded and ready for testing.

Notes for testing:
- To insert dummy data: delete projectmaestro.db -> uncomment line 6 in index.php -> run php -S and go to index page -> comment line 6
- Current dummy data might not be enough to test all cases, so feel free to add more in insertDummyData method of DatabaseHelper class (contact Database team if you need help on this)

Notes from previous pr which was replace by this one:
#PR79
What was done:
- logout will now go to logout.php (session and cookie clear) and then redirects back to index.php
- registrating with an existing email in the db will show an error
- UserId is now passed in as $_SESSION['instructor_id'] once user logins or registers successfully
- registration will now create sessions and go to home.php instead of login.php
- restriction access is put on home.php, if user not logged in, cannot access (this is only done on home.php atm, avoiding merging conflicts if pages are not done)

What testing team should check:
- registration on an existing email should show error
- once registered, redirects to home page
- try logging out and type in the header localhost:XXXX/app/home.php You should be redirected back to index.php

CalvinLoungsay committed:
Uses db file directly

CalvinLoungsay committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

chrisbrzhang committed:
initialized db at the top of all instructors pages

chrisbrzhang committed:
initialized db at the top of all instructor pages

Bywfaru committed:
Added ProjectOutline table to the database

MichaelPCLin committed:
Edited view_team, view_teams, check_courses, w8ing on db

chrisprosan committed:
Merge pull request #76 from medhatelmasry/backend_dbteam

Added ProjectOutline to the database schema

Bywfaru committed:
Fixed index.php not correctly initializing a database connection

Bywfaru committed:
Commented out insertDummyData() function call

Bywfaru committed:
Ignoring test db files

Bywfaru committed:
Fixed get functions not returning data

Bywfaru committed:
'View Projects' and view_projects.php works properly now

chrisprosan committed:
Merge pull request #77 from medhatelmasry/backend_dbteam

Fixed issue with registration_validation.php not initializing the database connection properly and made 'view_projects.php' functional

howareyouhelen committed:
logout clear session. add in userid in session. restricted access on home page. registration restriction on dup email.

howareyouhelen committed:
fix session start header error.

howareyouhelen committed:
fix on registration validation for instructor existing email check.

MichaelPCLin committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

MichaelPCLin committed:
Did some changes to check projects.php, calvn pls edit this based on your end. view_teams.php, view_team.php updated, they use session variables for the back buttons. add_members.php is entirely untested, bugs expected. However, the logic on the page should be correct. Syntax may cause some problems. view_teams.php, view_team.php, add_member.php require checks on the sql queries to db. Comments have been added to show where the sql calls go.

Bywfaru committed:
Updated database schema

Bywfaru committed:
Merge branch 'backend' into backend_dbteam

chrisprosan committed:
added empty view_project.php

chrisprosan committed:
Merge pull request #84 from medhatelmasry/backend_dbteam

Updated database schema

Bywfaru committed:
Fixed ProjectMember dummy data

oves98 committed:
used sessions and set validations

oves98 committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

Bywfaru committed:
Fix 2

Bywfaru committed:
Removed TeamId in create ProjectMember

Bywfaru committed:
Removed all traces of TeamId

chrisprosan committed:
Merge pull request #85 from medhatelmasry/backend_dbteam

Fixed ProjectMember dummy data

CalvinLoungsay committed:
Pages updated

chrisprosan committed:
changed InstructorId and StudentId to INTEGER AUTOINCREMENT
Fixed CourseId type in ProjectOutline (INTEGER -> VARCHAR)
Fixed dummy data
Note: To insert dummy data: delete projectmaestro.db -> uncomment line 6 in index.php -> run php -S and go to index page -> comment line 6

Added redirect from check_projects to create_project_outline
Moved sanitize_input function in processCreate to sanitize_input.php (for reuse)
Added functionality to create_project_outline.php
Added functionality to view_projects.php
- Sends projectId through GET to view_project.php
- Assumes ProjectOutlineId = 1, since check_projects.php does not send ProjectOutlineId at the moment

chrisprosan committed:
Changed InstructorId and StudentId back to VARCHAR. Should be INTEGER but I realized it's too late for that

Bywfaru committed:
Removed debugging echoes

CalvinLoungsay committed:
Course info passed to other pages and outline id

chrisprosan committed:
In Course table, changed InstructorId to UserId

oves98 committed:
passed courseId along to the pages that need them

oves98 committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

oves98 committed:
added a validation so the instructor can only see thier own course

chrisprosan committed:
hashed dummy password and insert to database

chrisprosan committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

oves98 committed:
fixed bug in add student page

chrisprosan committed:
fixed check_projects.php so it will only display project outlines belong to a specific course

chrisprosan committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

chrisprosan committed:
passed courseId to view_project.php

MichaelPCLin committed:
Refactored view_team, view_goals coded, add_members still in testing

MichaelPCLin committed:
Refactored view_team, view_goals coded, add_members still in testing

MichaelPCLin committed:
Added view_goals.php

chrisprosan committed:
fixed redirect to View Project page

chrisprosan committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

Bywfaru committed:
Fixed function in the helper

Bywfaru committed:
Merge

Bywfaru committed:
Undid jwt_secure.php vendor path change

chrisprosan committed:
Merge pull request #88 from medhatelmasry/backend_dbteam

Fix function

Bywfaru committed:
viewCourse.php was throwing an error because it was missing a boolean check on

Bywfaru committed:
Fixed it so that it displays the table still if there are no results

Bywfaru committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend_dbteam

Bywfaru committed:
Fixed errors that were being thrown

oves98 committed:
added logout to nav bar in create and view courses

oves98 committed:
added logout to nav bar in create and view courses

chrisprosan committed:
fixed transition from create_project_outline to check_projects

chrisprosan committed:
resolved merge conflicts

chrisprosan committed:
Merge pull request #89 from medhatelmasry/backend_dbteam

Fixed Michael's issue along with others. See commits

chrisprosan committed:
fixed view course page

chrisprosan committed:
Added functionality and linking to all instructor pages

ajytan reviewed (changes requested):
Bugs have been found. **Please look at issues _91 to 99_**.

oves98 committed:
fixed issues with delete paegs

oves98 committed:
added headers to create course pages

chrisprosan committed:
- deleted unused files
- restricted access to create_project_outline and view_projects
- fixed a bug in create_project_outline where courseId has an extra trailing slash
- fixed nav bar for create_project_outline and view_projects
- added back button from create_project_outline and view_projects to check_projects

chrisprosan committed:
resolved conflict

MichaelPCLin committed:
added page reliance on session variables, otherwise redirect to view_team.php, view_goal.php, add_member.php

CalvinLoungsay committed:
Navbars fixed and back buttons added

CalvinLoungsay committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

MichaelPCLin committed:
added some variable validation to view_goals.php, view_team.php and add_member.php, removed depreciated view_teams.php. Pages now have some simple errors for sql query fails and automatically redirect to index.php if no session variables are present.

CalvinLoungsay committed:
Added success message for adding a student

CalvinLoungsay committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

Bywfaru committed:
Added a container div to home, login, viewCourse, and index to make thing slook better

mikehuang123 committed:
styling

add some styling -> <div class = container></div>

Bywfaru committed:
Fixing API not setting primary key and key properly

mikehuang123 committed:
styling

more div container add

Bywfaru committed:
Commented out/removed debug code and added some documentation

oves98 committed:
added restrictions to pages so it cant be access throughthe url

chrisprosan committed:
Merge branch 'backend' into backend_dbteam

chrisprosan committed:
Merge pull request #109 from medhatelmasry/backend_dbteam

Fixed api.php to properly set the primary key and key value from the exploded path

chrisprosan committed:
put add buttons in div container; fixed not able to reach view goals page

chrisprosan committed:
added body and html tag for viewCourse, createCourse and delete

ASingh368 reviewed (approved):
all pages such that require the user to login cannot be accessed by modifying the url. Everything works!
**Tested Features:**
* registering with an existing email
* URL blocking
* Button Functionality

CalvinLoungsay committed:
Reduced the large empty space when no project outlines

CalvinLoungsay committed:
Merge branch 'backend' of https://github.com/medhatelmasry/Project-Maestro into backend

ajytan reviewed (approved):
**Tested Features:**
- Tested functionality (create course, add student, create project outline, delete course)
- Tested URL blocking
- Tested login and registration improvements

Tested functionality and styling of the site.
